streams/src/test/java/org/apache/kafka/streams/integration/AbstractResetIntegrationTest.java

```diff
@@ -109,7 +109,11 @@ static void afterClassGlobalCleanup() {
     void beforePrepareTest() throws Exception {
         ++testNo;
         bootstrapServers = cluster.bootstrapServers();
-        mockTime = cluster.time;
+
+        // we align time to seconds to get clean window boundaries and thus ensure the same result for each run
+        // otherwise, input records could fall into different windows for different runs depending on the initial mock time
+        final long alignedTime = (System.currentTimeMillis() / 1000) * 1000;
+        mockTime = new MockTime(0, alignedTime, System.nanoTime());
```

Comment:
This seems to overlap with #4095 -- should not be part of this PR IMHO.

Reply:
Yup, it was included just trying to see if the Jenkins job would be more stable then. Will rebase once 4095 is merged in.